backend/src/main/java/com/springBoot/fruits_ecommerce/services/JwtService.java

```diff
@@ -0,0 +1,80 @@
+package com.springBoot.fruits_ecommerce.services;
+ 
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.crypto.SecretKey;
+
+import org.springframework.stereotype.Service;
+
+import com.springBoot.fruits_ecommerce.models.Role;
+import com.springBoot.fruits_ecommerce.models.User;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+
+@Service
+public class JwtService {
+    private final String secret="12df32e69496a167c9d6196abec4bd4a25858e8da5fa4fe150ac9e03431ca36b";
+    public <T> T extractClaim(String token, Function<Claims, T> claimsResolver) {
+        final Claims claims = extractAllClaims(token);
+        return claimsResolver.apply(claims);
+    }
+
+    public String extractEmail(String token) {
+        return extractClaim(token, Claims::getSubject);
+    }
+
+    public Date extractExpiration(String token) {
+        return extractClaim(token, Claims::getExpiration);
+    }
+    private Claims extractAllClaims(String token) {
+        return Jwts.parser().verifyWith( getSigningKey()).build().parseSignedClaims(token).getPayload();
+    }
+
+    private Boolean isTokenExpired(String token) {
+        return extractExpiration(token).before(new Date());
+    }
+     public String generateToken(User user) {
+
+        Map<String, Object> extraClaims = addExtraClaims(user);
+    
+        return createToken( user,extraClaims );
+    }
+    private Map <String, Object>addExtraClaims(User user ){
+          Map<String, Object> extraClaims = new HashMap<>();
+        extraClaims.put("roles", user.getRoles().stream()
+            .map(Role::getName)
+            .collect(Collectors.toList()));
+
+        return extraClaims;
+
+    }
+     public String generateToken(User user,Map<String, Object> extraClaims) {
+    
+        return createToken( user,extraClaims);
+    }
+ 
+     private String createToken( User user, Map<String, Object> extraClaims) {
+        return Jwts.builder().claims(extraClaims).subject(user.getEmail()).issuedAt(new Date(System.currentTimeMillis()))
+                .expiration(new Date(System.currentTimeMillis() + 1000 * 60 * 60 * 10))
```

Comment:
Let's move the expiration period "1000 * 60 * 60 * 10" to be environment variable